File: app/src/main/java/com/coder/zzq/smartshowdemo/TestTypeToastActivity.java
package com.coder.zzq.smartshowdemo;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import com.coder.zzq.smartshow.toast.SmartToast;

public class TestTypeToastActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_test_type_toast);
    }

    public void onInfoClick(View view) {
        SmartToast.info(R.string.net_fine);
    }

    public void onSuccessClick(View view) {
        SmartToast.success(R.string.delete_succ);
    }

    public void onErrorClick(View view) {
        SmartToast.error(R.string.data_parse_error);
    }

    public void onWarningClick(View view) {
        SmartToast.warning(R.string.power_low);
    }

    public void onFailClick(View view) {
        SmartToast.fail(R.string.save_fail);
    }

    public void onCompleteClick(View view) {
        SmartToast.complete(R.string.download_complete);
    }


    public void onForbidClick(View view) {
        SmartToast.forbid(R.string.forbid_op);
    }

    public void onWaitingClick(View view) {
        SmartToast.waiting(R.string.wait_to_download);
    }


    public void onNextPageClick(View view) {
        startActivity(new Intent(this, TestTypeToastActivity.class));
    }


}

File: app/src/main/java/com/coder/zzq/smartshowdemo/TestTypeToastActivity.kt
package com.coder.zzq.smartshowdemo

import android.os.Bundle
import android.widget.AdapterView
import androidx.appcompat.app.AppCompatActivity
import com.coder.vincent.smart_dialog.click_list.ClickListAdapter
import com.coder.vincent.smart_toast.SmartToast
import com.coder.zzq.smartshowdemo.databinding.ActivityTestTypeToastBinding

class TestTypeToastActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(ActivityTestTypeToastBinding.inflate(layoutInflater).apply {
            val adapter = ClickListAdapter()
            adapter.setItemCenter(true)
            adapter.setItems(
                listOf(
                    "info",
                    "warning",
                    "success",
                    "error",
                    "fail",
                    "complete",
                    "forbid",
                    "waiting"
                )
            )
            listView.adapter = adapter
            listView.onItemClickListener = AdapterView.OnItemClickListener { _, _, position, _ ->
                when (position) {
                    0 -> onInfoClick()
                    1 -> onWarningClick()
                    2 -> onSuccessClick()
                    3 -> onErrorClick()
                    4 -> onFailClick()
                    5 -> onCompleteClick()
                    6 -> onForbidClick()
                    7 -> onWaitingClick()
                }
            }
        }.root)
    }

    private fun onInfoClick() {
        SmartToast.emotion().info(R.string.net_fine)
    }

    private fun onSuccessClick() {
        SmartToast.emotion().success(R.string.delete_succ)
    }

    private fun onErrorClick() {
        SmartToast.emotion().error(R.string.data_parse_error)
    }

    private fun onWarningClick() {
        SmartToast.emotion().warning(R.string.power_low)
    }

    private fun onFailClick() {
        SmartToast.emotion().fail(R.string.save_fail)
    }

    private fun onCompleteClick() {
        SmartToast.emotion().complete(R.string.download_complete)
    }

    private fun onForbidClick() {
        SmartToast.emotion().forbid(R.string.forbid_op)
    }

    private fun onWaitingClick() {
        SmartToast.emotion().waiting(R.string.wait_to_download)
    }
}